### ISSUE TITLE
Failing to Parse Valid T-SQL JSON Statement

### ISSUE THREAD

**Issue by rkelly** [sqlfluff/sqlfluff#3772]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff appears to not be able to parse this valid T-SQL query.

*min.sql*
```sql
SELECT (
    SELECT 1 AS a
    FOR JSON PATH, WITHOUT_ARRAY_WRAPPER
    )
```

```
$ sqlfluff lint min.sql
== [min.sql] FAIL
L:   3 | P:   9 |  PRS | Line 3, Position 9: Found unparsable section: 'FOR JSON
                       | PATH, WITHOUT_ARRAY_WRAPPER'
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### Expected Behaviour

I'd expect no errors.

```
All Finished 📜 🎉!
```
If I comment out the JSON line, I do get no errors.

### Observed Behaviour

See above

### How to reproduce

See above

### Dialect

tsql

### Version

```
$ sqlfluff --version
sqlfluff, version 1.3.0
$ python --version
Python 3.10.6
```

### Configuration

*.sqlfluff*
```
[sqlfluff]
dialect = tsql
exclude_rules = L003, L014, L016, L027, L031, L034, L051, L059, L061, L065
large_file_skip_char_limit = 80000

[sqlfluff:rules]
tab_space_size = 4

max_line_length = 85
indent_unit = space
comma_style = leading

single_table_references = consistent

aliasing = explicit
allow_scalar = False
force_enable = False
fully_qualify_join_types = outer
prefer_quoted_identifiers = False

[sqlfluff:indentation]
indented_joins = True
indented_ctes = False
indented_using_on = true
template_blocs_indent = false
disallow_hanging_indents = True

[sqlfluff:rules:L003]
hanging_indents = True
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
Just triaging this and yet, the `FOR JSON` syntax hasn't yet been implemented in our `TSQL` grammar. Docs here: https://docs.microsoft.com/en-us/sql/relational-databases/json/format-query-results-as-json-with-for-json-sql-server?view=sql-server-ver16
